### PR TITLE
Refactor PlayButton and remove GlobalPlayButton

### PR DESCRIPTION
**Pull request by MrTimscampi**
GlobalPlayButton essentially duplicated what PlayButton was doing already, while adding shuffle
support.

This removes that component and adds the same functionality to PlayButton, to avoid having
multiple components with the same purpose.

Blocked by #841 since the tests need some changes (and added cases) but that PR changes the spec file a lot.

### CODE REVIEW

File: pages/library/_viewId.vue [hunk 1]
```diff
@@ -34,17 +34,8 @@
         @change="onChangeFilter"
       />
       <v-spacer />
-      <global-playback-button
-        v-if="isQueueable"
```
Comment (ferferga) [inline pages/library/_viewId.vue:38]: This computed property must be removed

File: pages/series/_itemId/index.vue [hunk 2]
```diff
@@ -32,19 +32,10 @@
               'ml-1': $vuetify.breakpoint.mdAndUp
             }"
           >
-            <v-btn
-              v-if="canPlay(item)"
```
Comment (ferferga) [inline pages/series/_itemId/index.vue:36]: canPlay can be removed

File: pages/musicalbum/_itemId/index.vue [hunk 3]
```diff
@@ -38,18 +38,10 @@
               'ml-1': $vuetify.breakpoint.mdAndUp
             }"
           >
-            <v-btn
-              v-if="canPlay(item)"
```
Comment (ferferga) [inline pages/musicalbum/_itemId/index.vue:42]: canPlay can be removed

**Review by ferferga — CHANGES_REQUESTED** [review]

File: pages/series/_itemId/index.vue (hunk 2 above)
Comment (MrTimscampi) [reply pages/series/_itemId/index.vue:36]: It's part of itemsHelper, which is necessary for other things at the moment (namely the appearances).

File: pages/musicalbum/_itemId/index.vue (hunk 3 above)
Comment (MrTimscampi) [reply pages/musicalbum/_itemId/index.vue:42]: Same here.

File: mixins/itemHelper.ts [hunk 4]
```diff
@@ -44,11 +44,9 @@ const itemHelper = Vue.extend({
...
      */
     canPlay(item: BaseItemDto): boolean {
-      const itemType = item.Type || '';
-      const itemMediaType = item.MediaType || '';
-
       if (
-        [
+        item &&
+        ([
           'Audio',
```
Comment (ferferga) [inline mixins/itemHelper.ts:50]: Non-blocking, but I preferred the old way of doing this, I think it was much easier to look at

**Review by ferferga — APPROVED** [review]

**Review by ThibaultNocchi — APPROVED** [review]

**Pull request merged**